### PR TITLE
Recommend TrustPermission over TrustContributors

### PR DESCRIPTION
**Pull request by jglick**
A more secure default, with better documentation of the effects of other choices.

@reviewbybees

### CODE REVIEW

File: src/main/resources/org/jenkinsci/plugins/github_branch_source/ForkPullRequestDiscoveryTrait/help-trust.html [hunk 1]
```diff
@@ -19,12 +19,13 @@
...
         <dt>Contributors</dt>
         <dd>
-            Pull requests from <a href="https://developer.github.com/v3/repos/collaborators/">collaborators</a>
+            Pull requests from <a href="https://developer.github.com/v3/repos/collaborators/" target="_blank">collaborators</a>
             to the origin repository will be treated as trusted, all other pull requests from fork repositories
             will be treated as untrusted.
             Note that if credentials used by Jenkins for scanning the repository does not have permission to
             query the list of contributors to the origin repository then only the origin account will be treated
             as trusted - i.e. this will fall back to <code>Nobody</code>.
+            <strong>NOTE:</strong> all collaborators are trusted, even if they are only members of team with read permission.
```
Comment (daniel-beck) [inline src/main/resources/org/jenkinsci/plugins/github_branch_source/ForkPullRequestDiscoveryTrait/help-trust.html]: Grammar

File: src/main/resources/org/jenkinsci/plugins/github_branch_source/ForkPullRequestDiscoveryTrait/TrustContributors/config.properties [hunk 2]
```diff
@@ -0,0 +1 @@
+blurb=<div class="warning">This option may be insecure in some environments. See help button.</div>
```
Comment (daniel-beck) [inline src/main/resources/org/jenkinsci/plugins/github_branch_source/ForkPullRequestDiscoveryTrait/TrustContributors/config.properties]: There must be a better phrasing than "See help button" for the concept, but I cannot think of one.

File: src/main/resources/org/jenkinsci/plugins/github_branch_source/ForkPullRequestDiscoveryTrait/help-trust.html [hunk 3]
```diff
@@ -35,8 +36,9 @@
         <dd>
             Pull requests forks will be treated as trusted if and only if the fork owner has either Admin or Write
             permissions on the origin repository.
+            This is the recommended policy.
```
Comment (stephenc) [inline src/main/resources/org/jenkinsci/plugins/github_branch_source/ForkPullRequestDiscoveryTrait/help-trust.html]: Consider using bold format for this sentence

**Comment by jglick** [comment]
@reviewbybees done

**Pull request merged**